### PR TITLE
(GH-39) Add fluent API to settings classes & aliases

### LINKED ISSUE #39: Add fluent API to settings classes
Some people prefere a fluent syntax for building the settings for an alias. We should add overloads to the aliases which an `Action` can be passed returning the concrete settings class. Additional there should be extensions on the settings classes which allow to build the settings object in a fluent syntax.

Similar to http://cakebuild.net/api/Cake.Common.Tools.MSBuild/MSBuildAliases/F36093FE and http://cakebuild.net/api/Cake.Common.Tools.MSBuild/MSBuildSettings/.

### PR DESCRIPTION
**Pull request by pascalberger**
Add fluent API to settings classes and aliases for passing and settings configurator.

Fixes #39

### CODE REVIEW

**Comment by pascalberger** [comment]
@Philo Any opinion about this? There seems to be some interested from users to have a fluent API for creating the settings objects.

File: src/Cake.Npm/Install/NpmInstallSettingsExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+            settings.Global = false;
+            return settings;
+        }
+
+        /// <summary>
+        /// Defines that npm should ignore modules listed in <c>devDependencies</c>.
+        /// </summary>
+        /// <param name="settings">The settings.</param>
+        /// <returns>The <paramref name="settings"/> instance with <see cref="NpmInstallSettings.Production"/> set to <c>true</c>.</returns>
+        public static NpmInstallSettings OnProduction(this NpmInstallSettings settings)
```
Comment (Philo) [inline src/Cake.Npm/Install/NpmInstallSettingsExtensions.cs]: Not convinced by the term `OnProduction`, perhaps something like `.ForProduction` or `.ProductionOnly` ?

**Comment by Philo** [comment]
Yes, I would also be keen to re-introduce the fluent api as I'm also find the creation of the settings object a bit tedious and "untidy".

Just added a minor comment now,. other than that I'm happy with adding this as it also doesn't introduce any breaking changes

File: src/Cake.Npm/Install/NpmInstallSettingsExtensions.cs (hunk 1 above)
Comment (pascalberger) [inline src/Cake.Npm/Install/NpmInstallSettingsExtensions.cs]: @Philo I renamed to `ForProduction`

**Pull request merged**